lib/AST/SubstitutionMap.cpp

```diff
@@ -226,6 +223,46 @@ SubstitutionMap SubstitutionMap::subst(TypeSubstitutionFn subs,
   return result;
 }
 
+void SubstitutionMap::toList(SmallVectorImpl<Substitution> &result) const {
+  auto *genericSig = getGenericSignature();
+  if (genericSig == nullptr)
+    return;
```

Comment:
To be on the safe side, do `result.clear()` before returning. We don't know if `result` was empty before the call.